### PR TITLE
feat: move region metadata to store-api

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- move [RegionMetadata] to `store-api`
- define its own error type
- remove the "version"

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/store-api/src/metadata.rs [hunk 1]
```diff
@@ -28,3 +43,673 @@ pub struct ColumnMetadata {
     /// Immutable and unique id of a region.
     pub column_id: ColumnId,
 }
+
+/// Initial version number of a new region.
+// pub(crate) const INIT_REGION_VERSION: VersionNumber = 0;
```
Comment (evenyag) [inline src/store-api/src/metadata.rs:48]:
```suggestion
```

File: src/datanode/src/region_server.rs [hunk 2]
```diff
@@ -94,10 +95,58 @@ impl RegionServer {
...
+//     async fn scan(
+//         &self,
+//         state: &SessionState,
+//         projection: Option<&Vec<usize>>,
+//         filters: &[Expr],
+//         limit: Option<usize>,
+//     ) -> Result<Arc<dyn DfPhysicalPlan>> {
+//         todo!()
+//     }
+// }
```
Comment (evenyag) [inline src/datanode/src/region_server.rs:152]: ......

File: src/store-api/Cargo.toml [hunk 3]
```diff
@@ -17,6 +18,7 @@ datatypes = { workspace = true }
 derive_builder.workspace = true
 futures.workspace = true
 serde.workspace = true
+serde_json = "1.0"
```
Comment (v0y4g3r) [inline src/store-api/Cargo.toml:21]:
```suggestion
serde_json.workspace = true
```

**Review by v0y4g3r — APPROVED** [review]
Mostly LGTM

**Review by evenyag — APPROVED** [review]

**Pull request merged**